### PR TITLE
[#4] Initial config refactor for write command

### PR DESCRIPTION
**Pull request by vaugenwake**
Basic working version of an init config command to simplify onboarding and connection to your jira account.

**Todo:**

- [ ] Add tests (I am not really sure how we are going to test this TBH)
- [x] Add form validation for inputs like email & URL
- [ ] Add a ping request to validate the config works and we can connect to the API

### CODE REVIEW

**Comment by vaugenwake** [comment]
Addresses #4

**Comment by vaugenwake** [comment]
@matthewrobinsondev @theoboldalex Thoughts on this? Not to sure about how I have handled the file paths etc.

Basically Viper doe not let you write the config if the file does not already exist, so needed to check for the directory and the file before I tried to write the config definition from Viper to the file.

Also @matthewrobinsondev whats your thoughts on how to approach testing this bad boy? Write a file in the test and assert it and then clean up the file after the test is closing off maybe?

File: pkg/config/config.go [hunk 1]
```diff
@@ -3,25 +3,77 @@ package config
...
+func (c ConfigService) getFullPath() string {
+	return path.Join(c.getBasePath(), c.getFilename())
+}
+
+func (c ConfigService) getFilename() string {
+	return CONFIG_NAME + "." + CONFIG_TYPE
+}
+
+func (c ConfigService) createIfNotExist() error {
+	if err := os.MkdirAll(c.getBasePath(), 0755); err != nil {
```
Comment (theoboldalex) [inline pkg/config/config.go]: const for the file permissions?

**Comment by matthewrobinsondev** [comment]
> Also @matthewrobinsondev whats your thoughts on how to approach testing this bad boy? Write a file in the test and assert it and then clean up the file after the test is closing off maybe?

Yah, create a tmp directory through os int he test, write & think` t.cleanup` allows you to do _the cleanup_

**Comment by vaugenwake** [comment]
Finished the basic working auth/setup command. Also refactored a few bits to be a bit more usable in the future.

Did look at adding tests for the auth bits but for the life of me can't find a decent way to deal with viper

**Comment by matthewrobinsondev** [comment]
going to create an issue for enhancement in the future & not for this PR but I think being able to show the config would be easy & useful

File: readme.md [hunk 2]
```diff
@@ -1,4 +1,25 @@
 # Work In Progress!
 On the quest to never leave the terminal.
 
-**Code before POs**
```
Comment (matthewrobinsondev) [inline readme.md:4]: smh

**Review by matthewrobinsondev — APPROVED** [review]

**Comment by matthewrobinsondev** [comment]
gonna merge, unless any objections

**Pull request merged**